add logstash output

thabiger opened the pull request:
**What type of PR is this?**

/kind feature

**Any specific area of the project related to this PR?**

/area outputs

**What this PR does / why we need it**:

This PR adds the ability to write Falco events to Logstash, enabling users to leverage the ELK stack for processing and visualizing Falco events. Although Falcosidekick can already communicate directly with Elasticsearch, this integration allows users to utilize their existing ELK stacks without changing their workflow.

**Special notes for your reviewer**:

We've been using this output in a production environment for several months now, and it has proven its reliability.

poiana commented:
Welcome @thabiger! It looks like this is your first PR to falcosecurity/falcosidekick 🎉

leogr commented:
> @leogr @cpanato Is there anything I can do to help with the processing? If there’s any reason it might not be approved, knowing that would be really helpful too :)

Hey @thabiger 

I apologize for the delay. We have just been busy. You already did everything you could, and we appreciate this. :pray: 

I did a quick look, and it seems good to me. @cpanato can you give a second look? Then we can approve.

Thank you, and again, sorry for this taking so long.

leogr commented on outputs/logstash.go:
```diff
@@ -0,0 +1,137 @@
+	c.PromStats.Outputs.With(map[string]string{"destination": "logstash", "status": status}).Inc()
+	go c.CountMetric(Outputs, 1, []string{"output:logstash", fmt.Sprintf("status:%v", status)})
+
+	log.Printf("[%v] : output.logstash status=%v (%v)\n", loglevel, status, n)
```
Is this a leftover?

thabiger replied:
Aside from switching to log.Printf, which I’ve already updated, the rest looks fine — logLevel and status are already being set above based on the err.

leogr commented on outputs/logstash.go:
```diff
@@ -0,0 +1,137 @@
+
+		cert, err := tls.LoadX509KeyPair(MutualTLSClientCertPath, MutualTLSClientKeyPath)
+		if err != nil {
+			log.Printf("[ERROR] : Logstash - Failed to load SSL certificate: %v\n", err)
+		}
+
+		caCert, err := os.ReadFile(MutualTLSClientCaCertPath)
+		if err != nil {
+			log.Printf("[ERROR] : Logstash - Failed to load SSL CA certificate: %v\n", err)
```
Should we use `utils.Log()` instead?

thabiger replied:
Switched to utils.Log() as suggested.

thabiger commented:
> Hey @thabiger
> 
> Can you take a look at the above suggestions? Otherwise SGTM.

Sorry for the delay — I’ve been swamped with work away from the computer lately. I’ll definitely take a look by tomorrow. Thanks a lot for the review!

poiana commented:
Thanks for your pull request. Before we can look at it, you'll need to add a 'DCO signoff' to your commits.

:memo: **Please follow instructions in the [contributing guide](https://github.com/falcosecurity/.github/blob/main/CONTRIBUTING.md) to update your commits with the DCO**

Full details of the Developer Certificate of Origin can be found at [developercertificate.org](https://developercertificate.org/).

**The list of commits missing DCO signoff**:

- [ed0c4cd](https://github.com/falcosecurity/falcosidekick/commits/ed0c4cd09298130f94e47fa576a7a7dd4e43b807) address review comments

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

thabiger commented on config.go:
```diff
@@ -414,6 +414,18 @@ var outputDefaults = map[string]map[string]any{
+	"Logstash": {
+		"Address":         "",
+		"Port":            5044,
+		"Password":        "",
```
It was unnecessary. I removed it in the latest commit.

thabiger commented:
Hi @leogr 

I’ve fixed all the issues in the latest commit. I also rebased the code on top of master, built it locally, and successfully ran basic tests in my local environment - everything worked fine :)

poiana commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/falcosecurity/falcosidekick/pull/1146#pullrequestreview-2912624619" title="Approved">leogr</a>*, *<a href="https://github.com/falcosecurity/falcosidekick/pull/1146#" title="Author self-approved">thabiger</a>*

The full list of commands accepted by this bot can be found [here](https://prow.falco.org/command-help?repo=falcosecurity%2Ffalcosidekick).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/falcosecurity/falcosidekick/blob/master/OWNERS)~~ [leogr]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->

poiana commented:
LGTM label has been added.  <details>Git tree hash: 81258c4d9af4eb09758205cb27963e6279b93663</details>
